### PR TITLE
Remove deprecated `@NativeImageTest`

### PR DESCRIPTION
**Pull request by gastaldi**
- This finally removes the long-deprecated `@NativeImageTest` (since 2.8.0.Final) and its reference in the documentation.
- It also updates the migration script to support changing from `@NativeImageTest` to `@QuarkusIntegrationTest`
- Depends on https://github.com/quarkusio/quarkus-quickstarts/pull/1239

### CODE REVIEW

**Comment by geoand** [comment]
+1 from me, but I would like more people to weigh in

**Comment by gastaldi** [comment]
@gsmet @maxandersen thoughts?

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -277,7 +277,7 @@ to 300 seconds, use: `./mvnw verify -Pnative -Dquarkus.test.wait-time=300`.
 
 [WARNING]
 ====
-This procedure was formerly accomplished using the `@NativeImageTest` annotation. `@NativeImageTest` is considered deprecated in favor of `@QuarkusIntegrationTest` which provides a superset of the testing
+This procedure was formerly accomplished using the `@NativeImageTest` annotation. `@NativeImageTest` was replaced by `@QuarkusIntegrationTest` which provides a superset of the testing
```
Comment (zakkak) [inline docs/src/main/asciidoc/building-native-image.adoc:280]: Not sure if this still worth keeping around.

**Review by zakkak — CHANGES_REQUESTED** [review]
@gastaldi it looks like there are still some references to NativeImageTest left in the code:

```java
core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java
87:    public static final DotName NATIVE_IMAGE_TEST = DotName.createSimple("io.quarkus.test.junit.NativeImageTest");

core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestConfig.java
180:     * Used in {@code @QuarkusIntegrationTest} and {@code NativeImageTest} to determine how long the test will wait for the
```

You might also want to rename the following variables
```java
core/runtime/src/main/java/io/quarkus/runtime/test/TestScopeSetup.java
5:    void setup(boolean isNativeImageTest);
7:    void tearDown(boolean isNativeImageTest);

test-framework/common/src/main/java/io/quarkus/test/common/TestScopeManager.java
19:    public static void setup(boolean isNativeImageTest) {
21:            i.setup(isNativeImageTest);
25:    public static void tearDown(boolean isNativeImageTest) {
27:            i.tearDown(isNativeImageTest);

extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ArcTestRequestScopeProvider.java
14:    public void setup(boolean isNativeImageTest) {
15:        if (isNativeImageTest) {
28:    public void tearDown(boolean isNativeImageTest) {
29:        if (isNativeImageTest) {

```

**Comment by gastaldi** [comment]
@zakkak good catch! Fixed and pushed

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 1 above)
Comment (gastaldi) [reply docs/src/main/asciidoc/building-native-image.adoc:280]: I kept it so people using `@NativeImageTest` in legacy references could understand why that annotation doesn't exist anymore

Comment (zakkak) [reply docs/src/main/asciidoc/building-native-image.adoc:280]: Sure, but I would expect this to be necessary only during the deprecation face. Anyway, it's OK with me.

**Review by zakkak — APPROVED** [review]
LGTM, thanks @gastaldi !

**Comment by rsvoboda** [comment]
@gastaldi - Needs breaking change label on this PR and entry in https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0

**Comment by rsvoboda** [comment]
Also `docs/src/main/asciidoc/building-native-image.adoc` mentions `@NativeImageTest`.

That doc should be also updated

**Comment by gastaldi** [comment]
> Also `docs/src/main/asciidoc/building-native-image.adoc` mentions `@NativeImageTest`.
> 
> That doc should be also updated

@rsvoboda as I explained in https://github.com/quarkusio/quarkus/pull/31048#discussion_r1107068819 I left it in purpose

**Comment by rsvoboda** [comment]
ok

**Comment by gastaldi** [comment]
Added to migration guide: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0#removal-of-ioquarkustestjunitnativeimagetest-and-ioquarkustestjunitdisabledonnativeimagetest-annotations

**Pull request merged**